xdsl/dialects/riscv.py

```diff
@@ -2689,25 +2689,38 @@ class GetFloatRegisterOp(GetAnyRegisterOperation[FloatRegisterType]):
 # region RISC-V Extensions
 
 
+class ScfgwOpHasCanonicalizationPatternsTrait(HasCanonicalisationPatternsTrait):
+    @classmethod
+    def get_canonicalization_patterns(cls) -> tuple[RewritePattern, ...]:
+        from xdsl.transforms.canonicalization_patterns.riscv import (
+            ScfgwOpUsingImmediate,
+        )
+
+        return (ScfgwOpUsingImmediate(),)
+
+
 @irdl_op_definition
 class ScfgwOp(RsRsIntegerOperation):
```

Comment:
```suggestion
class ScfgwOp(RdRsRsOperation[IntRegisterType, IntRegisterType, IntRegisterType]):
```

Might as well fix this, there is a result register, it's just constrained to be zero. We should similarly omit the rd when printing assembly as in `scfgwi`